### PR TITLE
Outfit datums the beginginging

### PR DESCRIPTION
**Pull request by ShiftyRail**
Or "how I learned to love lists and hate BYOND"

This is just a stretch/proof of concept of "outfit datums", a way to equip people (whether in the context of joining the game, spawning as a bomberman, or a wizard...) that doesn't suck ass.

Right now, it doesn't handle PDA/IDs, headsets (which will be a tad complicated to address), and the only job that's ported from the old tree of `equip_to_slot_or_del` to the outfit datums is Assistant.
However, alt-titles, custom species and all that junk is correctly handled.

Spawning the concrete air tank is still delegated to the species, although that can be changed.

As a technical note, no, you _can't_ have a list which associated a number to a type in BYOND.
Which is a bit annoying as it wraps annoying `text2num` and `num2text` everywhere.

Before this is merged, I would like to try and solve the problems of ID cards, headsets, and maybe make a proof of concept for ERTs, Death Commandos and Syndie Death Commandos as well.

In a later PR, I will allow the creator of a custom strike team to select which outfit the leader of the team should be given and which outfit the other members should have.

### CODE REVIEW

File: code/game/striketeams/striketeam_datums.dm [hunk 1]
```diff
@@ -346,6 +346,6 @@ var/list/sent_strike_teams = list()
 		customsquad = ticker.mode.CreateFaction(/datum/faction/strike_team/custom)
 		if(customsquad)
 			customsquad.HandleNewMind(new_commando.mind) //First come, first served
-	new_commando.equip_death_commando(leader_selected)
+	new_commando.equip_death_commando()
```
Comment (ShiftyRail) [inline code/game/striketeams/striketeam_datums.dm:349]: Whoops

**Comment by MadmanMartian** [comment]
New branch made for outfit datums, called outfit_datums.

**Comment by ShiftyRail** [comment]
It can be merged to live as in, there are provisions for both systems right now.
However, that's a bit schizophrenic so I don't mind you wanting it to be fully packaged before merging to live.

**Comment by ghost** [comment]
Yes
YES
God bless you Shifty

**Pull request merged**